### ISSUE TITLE
Dark mode

### ISSUE THREAD

**Issue by pgodschalk** [nextcloud/desktop#1287]
Add a dark mode that also respects system wide settings.

Related PR: #1439

---

#### macOS
It'd be great if the systemwide dark mode setting for macOS Mojave was respected. This feature is available in Qt 5.12: https://bugreports.qt.io/browse/QTBUG-68891

**Comment by skewty** [comment]
Intelligent dark mode switching on Linux would also be nice.

**Comment by jancborchardt** [comment]
@pgodschalk @skewty @yoshi2889 @artem-sidorenko @LPkkjHD dark mode is being worked on in the pull request at https://github.com/nextcloud/desktop/pull/1439 → it would be great if you could give it a test run and provide feedback! :)

**Comment by misch7** [comment]
> @pgodschalk @skewty @Yoshi2889 @artem-sidorenko @LPkkjHD dark mode is being worked on in the pull request at #1439 → it would be great if you could give it a test run and provide feedback! :)

And please have a look at our 2.6.1 RC1, it has a first preview of a mostly working dark mode:
https://download.nextcloud.com/desktop/prereleases/Mac/Nextcloud-qt5.12.5-2.6.1.20191018rc1.pkg

**Comment by artem-sidorenko** [comment]
![Screen Shot 2019-10-30 at 08 58 31](https://user-images.githubusercontent.com/1645670/67839295-9973a280-faf3-11e9-91b6-439ed398b3c8.png)
@jancborchardt @misch7 looks very promising for me :) many thanks!

**Comment by Yoshi2889** [comment]
Sorry for my late reply, I've only gotten around to testing it today.

It looks great and the overall app fits in better with the macOS theme in my opinion :)

The only issue I have is that some icons aren't very visible, like in the Share Options dialog. Other than that it really looks great though :)

**Comment by jancborchardt** [comment]
@artem-sidorenko @Yoshi2889 thanks so much for your screenshots, it helps a lot in finding and fixing the issues! :)

@misch7 let me know if you need any help with the icons, but I think we did that during the meeting? The activity icon, then confirm.svg, public.svg, add.svg

**Comment by rakekniven** [comment]
<img width="801" alt="Bildschirmfoto 2019-11-18 um 19 00 52" src="https://user-images.githubusercontent.com/2069590/69123209-0fd74500-0aa1-11ea-9d29-ba2bb5ec6d62.png">

The activity icon (the flash) is to dark IMHO

**Comment by jancborchardt** [comment]
@rakekniven of course in dark mode the activity icon should be white, as should the 3-dot menu icon. :) And all icons in the share dialog as well: confirm, share link, plus icon.

**Comment by claell** [comment]
@pgodschalk I edited your description and title, so this issue reflects the request for a dark mode in general, since a request for Windows and also Linux came up.

**Comment by tmechen** [comment]
![Bildschirmfoto 2020-07-30 um 16 07 17](https://user-images.githubusercontent.com/9123876/88932940-0da7c600-d27f-11ea-81b8-996d553b180a.png)


there is a coloring problem in osx dark mode: highlighted text needs another color maybe ;)

**Comment by rakekniven** [comment]
@tmechen which version are you using?

**Comment by tmechen** [comment]
Version 2.6.2stable (build 20191224). 
sry, just saw there is a newer one (did i miss an upgrade notification?)
will try again with latest

**Comment by tmechen** [comment]
okay, updateded to 2.6.5(macOS) but the problem still exists :( 

![Bildschirmfoto 2020-07-30 um 20 31 11](https://user-images.githubusercontent.com/9123876/88960363-ad774b00-d2a3-11ea-83ff-2ddb1ddef6ec.png)


this is the error message while trying to set up folder sync when chosing a non-empty folder
i inverted the colors of the image to read the message: 
![Bildschirmfoto 2020-07-30 um 20 33 04](https://user-images.githubusercontent.com/9123876/88960497-e31c3400-d2a3-11ea-80fb-724bce6e11e9.png)

**Comment by rakekniven** [comment]
@tmechen And which Mac OS version?

**Comment by tmechen** [comment]
yesterday i tested both with the macOS Catalina 10.15.5 and the newest update 10.15.6

**Comment by jancborchardt** [comment]
So what are the things left to do for dark mode in version 3.0?

**Comment by er-vin** [comment]
@misch7 we should be good on the widget side right?

On the QML side I expect that it just doesn't handle it at all, AFAICT the palette is enforced there.

**Comment by misch7** [comment]
@er-vin There are some minor widget problems left, for example in the share widget like shown in this comment: https://github.com/nextcloud/desktop/issues/1287#issuecomment-549046296
I didn't yet figure out why the colours are not getting adopted by the embedded child widget under some circumstances, maybe you have an idea? ;-)

Regarding the QML side you are right. Some time ago I did experiments with `theme/Style/Style.qml` but some hard-coded colour values have to be moved and implemented there, plus some logic for automatic dark/light contrast font colour picking, like I once did in the Theme class (background colour detection for widgets, also for correctly colouring error fonts on red backgrounds etc.).

**Comment by er-vin** [comment]
> @er-vin There are some minor widget problems left, for example in the share widget like shown in this comment: [#1287 (comment)](https://github.com/nextcloud/desktop/issues/1287#issuecomment-549046296)
> I didn't yet figure out why the colours are not getting adopted by the embedded child widget under some circumstances, maybe you have an idea? ;-)

Making a guess: there's a CSS on said widget or one of its parent?

This tends to break styles and palettes in "interesting" ways (no I'm not a huge fan of the CSS in widgets, beside I suspect this feature is barely maintained, handle with care).
 
> Regarding the QML side you are right. Some time ago I did experiments with `theme/Style/Style.qml` but some hard-coded colour values have to be moved and implemented there, plus some logic for automatic dark/light contrast font colour picking, like I once did in the Theme class (background colour detection for widgets, also for correctly colouring error fonts on red backgrounds etc.).

OK, let's ignore that for now and focus on fixing the remaining widget palette issues. The QML code will need some cleanup anyway before we can properly handle a dark mode there.

**Comment by elsiehupp** [comment]
I don't know if this issue is stale, but while the settings window does support Dark Mode on macOS now, the popover does not, and the settings window does not support Dark Mode on Ubuntu, or, if it does, it doesn't adequately detect when Dark Mode is on or not.

**Comment by x00-R0n1n** [comment]
still having this issue:
OSX: 10.15.7
Nextcloud Client: 3.2.1

This wasn't an issue till now, but its unreadable without changing my entire theme and Im not doing that.

Please fix!

Happens when attempting to setup folder sync for new item:

Settings > Add folder sync connection:

<img width="620" alt="image" src="https://user-images.githubusercontent.com/22737601/117890794-a9cd1780-b26a-11eb-9bdc-c1a67f5ef0c5.png">

**Comment by elsiehupp** [comment]
I just checked, and I’m having the same issue as @x00-R0n1n on macOS 11.3.1 with Nextcloud 3.2.1, and I hadn’t had this issue previously, so it’s probably new.

**Comment by rakekniven** [comment]
> still having this issue:
> OSX: 10.15.7 OS Dark Theme enabled
> Nextcloud Client: 3.2.1

This issue was about creating a dark mode.
That has been done.

Please open a new issue specific for your issue.

@er-vin Do you agree to close this general issue here?

**Comment by FlexW** [comment]
@rakekniven I will close this issue. @x00-R0n1n if the issue is new, please create a new one. This way we can better track what is a problem. @elsiehupp there is some ongoing work to make the main dialog respecting the system theme https://github.com/nextcloud/desktop/pull/3026
